feat: reply to invalid config

gitcoindev opened the pull request:
The bot replies directly to invocations with invalid config to make it clear why it's down. A new helper method upsertLastCommentToIssue was introduce to avoid adding multpile comments with the same error message.

Resolves: #799

Quality Assurance: https://github.com/gitcoindev/ubiquibot/issues/12

gitcoindev committed:
feat: reply to invalid config

The bot replies directly to invocations with invalid config to make it
clear why it's down. A new helper method upsertLastCommentToIssue was
introduce to avoid adding multpile comments with the same error message.

Resolves: #799

gitcoindev committed:
fix: add missing await for upsertLastCommentToIssue

Resolves: #799

cuddle-dev commented on src/helpers/issue.ts:
```diff
@@ -206,6 +206,25 @@ export const upsertCommentToIssue = async (issue_number: number, comment: string
+      per_page: 100,
+    });
+
+    if (comments.data[comments.data.length - 1].body !== commentBody) await addCommentToIssue(commentBody, issue_number);
```
```suggestion
    if (comments.data.length > 0 && comments.data[comments.data.length - 1].body !== commentBody) await addCommentToIssue(commentBody, issue_number);
```

cuddle-dev reviewed (dismissed):
Both codebase/ QA looking good

cuddle-dev committed:
chore: uUpdate src/helpers/issue.ts

cuddle-dev commented on src/helpers/issue.ts:
```diff
@@ -206,6 +206,25 @@ export const upsertCommentToIssue = async (issue_number: number, comment: string
+  const payload = context.payload as Payload;
+
+  try {
+    const comments = await context.octokit.issues.listComments({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      issue_number: issue_number,
+      per_page: 100,
+    });
```
Looks like it doesn't fetch all the comments. please use the `getAllIssueComments` method in src/helpers/issue.ts.

gitcoindev replied:
@0xcodercrane thank you for the quick review, I am on it.

gitcoindev committed:
feat: use getAllIssueComments in upsertLastCommentToIssue

As suggested during PR review, getAllIssueComments returns all issue
comments using paginated response.

Resolves: #799

gitcoindev committed:
feat: upsertLastCommentToIssue use raw format

gitcoindev commented:
@0xcodercrane I applied suggested changes and did a QA on issue with 182 comments:

https://github.com/gitcoindev/ubiquibot/issues/11#issuecomment-1757486779

gitcoindev commented:
Thank you for reviewing and merging!
